src/components/Products.jsx

```diff
@@ -1,33 +1,78 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-import {ProductCard, Grid} from './common'
+import {ProductCard, Grid, Button, NotFound} from './common'
 
 const Products = ({viewType, data}) => {
     return (
-        <section className='categories'>
-            <Grid col={4} mdCol={2} smCol={1} gap={20}>
-                {
-                    data?.results?.map((item, index) => 
-                        <ProductCard 
-                            key={index}
-                            img1={item.data.mainimage.url}
-                            img2={item.data?.images[0]?.image?.url}
-                            name={item.data.name}
-                            price={item.data.price}
-                            slug={item?.slugs[0]}
-                            alt={item.data.mainimage.alt}
-                        />   
-                    )
-                }
+        data.length > 0 ? renderProdcustGrid(viewType, data) :
+        <NotFound text='Products Not Found'/>
+    )
+}
+
+const renderProdcustGrid = (viewType, data) => {
+    return (
+        <>
+            <section className='categories'>
+                <Grid col={4} mdCol={2} smCol={1} gap={20}>
+                    {
+                        data?.map((item) => 
+                            <ProductCard 
+                                key={item.id}
+                                img1={item.data.mainimage.url}
+                                img2={item.data?.images[0]?.image?.url}
+                                name={item.data.name}
+                                price={item.data.price}
+                                slug={item?.slugs[0]}
+                                alt={item.data.mainimage.alt}
+                            />   
+                        )
+                    }
+                </Grid>
+            </section>
+
+            {
+                viewType === 'ProductList' && pagination()
```

Comment:
You can make use of component notation, would be more react friendly :), `<Pagination />`, and then you can reuse it and make more logic